diff --git a/third_party/curl.BUILD b/third_party/curl.BUILD
index 62fc946956d..c9d21172f66 100644
--- third_party/curl.BUILD
+++ third_party/curl.BUILD.new
@@ -510,7 +510,7 @@ genrule(
         "#  define HAVE_RAND_STATUS 1",
         "#  define HAVE_SSL_GET_SHUTDOWN 1",
         "#  define HAVE_TERMIOS_H 1",
-        "#  define OS \"x86_64-pc-linux-gnu\"",
+        "#  define OS \"$(CURL_OS)\"",
         "#  define RANDOM_FILE \"/dev/urandom\"",
         "#  define USE_OPENSSL 1",
         "#endif",
@@ -686,12 +686,12 @@ genrule(
         "#  define SEND_TYPE_ARG4 int",
         "#  define SEND_TYPE_RETV ssize_t",
         "#  define SIZEOF_INT 4",
-        "#  define SIZEOF_LONG 8",
+        "#  define SIZEOF_LONG $(CURL_SIZEOF_LONG)",
         "#  define SIZEOF_OFF_T 8",
         "#  define SIZEOF_CURL_OFF_T 8",
         "#  define SIZEOF_SHORT 2",
-        "#  define SIZEOF_SIZE_T 8",
-        "#  define SIZEOF_TIME_T 8",
+        "#  define SIZEOF_SIZE_T $(CURL_SIZEOF_LONG)",
+        "#  define SIZEOF_TIME_T $(CURL_SIZEOF_LONG)",
         "#  define SIZEOF_VOIDP 8",
         "#  define STDC_HEADERS 1",
         "#  define STRERROR_R_TYPE_ARG3 size_t",
