--- third_party/curl.BUILD	2019-09-23 11:34:08.733016562 -0700
+++ curl.BUILD	2019-09-23 11:33:33.815792309 -0700
@@ -268,7 +268,6 @@
             "-DCURL_DISABLE_NTLM",  # turning it off in configure is not enough
             "-DHAVE_LIBZ",
             "-DHAVE_ZLIB_H",
-            "-Wno-string-plus-int",
         ],
     }) + select({
         "@org_tensorflow//tensorflow:macos": [
@@ -418,7 +417,6 @@
             "-D_GNU_SOURCE",
             "-DHAVE_CONFIG_H",
             "-DCURL_DISABLE_LIBCURL_OPTION",
-            "-Wno-string-plus-int",
         ],
     }),
     deps = [":curl"],
@@ -493,7 +491,7 @@
         "#  define HAVE_RAND_STATUS 1",
         "#  define HAVE_SSL_GET_SHUTDOWN 1",
         "#  define HAVE_TERMIOS_H 1",
-        "#  define OS \"x86_64-pc-linux-gnu\"",
+        "#  define OS \"$(CURL_OS)\"",
         "#  define RANDOM_FILE \"/dev/urandom\"",
         "#  define USE_OPENSSL 1",
         "#endif",
@@ -669,12 +667,12 @@
         "#  define SEND_TYPE_ARG4 int",
         "#  define SEND_TYPE_RETV ssize_t",
         "#  define SIZEOF_INT 4",
-        "#  define SIZEOF_LONG 8",
+        "#  define SIZEOF_LONG $(CURL_SIZEOF_LONG)",
         "#  define SIZEOF_OFF_T 8",
         "#  define SIZEOF_CURL_OFF_T 8",
         "#  define SIZEOF_SHORT 2",
-        "#  define SIZEOF_SIZE_T 8",
-        "#  define SIZEOF_TIME_T 8",
+        "#  define SIZEOF_SIZE_T $(CURL_SIZEOF_LONG)",
+        "#  define SIZEOF_TIME_T $(CURL_SIZEOF_LONG)",
         "#  define SIZEOF_VOIDP 8",
         "#  define STDC_HEADERS 1",
         "#  define STRERROR_R_TYPE_ARG3 size_t",
