diff --git a/third_party/hwloc/BUILD.bazel b/third_party/hwloc/BUILD.bazel
index a9de93686c0..63e4d5b3b0e 100644
--- a/third_party/hwloc/BUILD.bazel
+++ b/third_party/hwloc/BUILD.bazel
@@ -209,14 +209,21 @@ _INCLUDE_PRIVATE_HWLOC_AUTOIGEN_CONFIG_H_LINUX_SUBS.update(_INCLUDE_PRIVATE_HWLO

 _INCLUDE_PRIVATE_HWLOC_AUTOIGEN_CONFIG_H_CUDA_SUBS.update(_INCLUDE_PRIVATE_HWLOC_AUTOIGEN_CONFIG_H_LINUX_SUBS)

+_INCLUDE_PRIVATE_HWLOC_AUTOIGEN_CONFIG_H_LINUX_ARMV7_SUBS = dict(_INCLUDE_PRIVATE_HWLOC_AUTOIGEN_CONFIG_H_LINUX_SUBS)
+_INCLUDE_PRIVATE_HWLOC_AUTOIGEN_CONFIG_H_LINUX_ARMV7_SUBS.update({
+    "#undef HWLOC_SIZEOF_UNSIGNED_LONG": "#define HWLOC_SIZEOF_UNSIGNED_LONG 4",
+    "#undef SIZEOF_UNSIGNED_LONG": "#define SIZEOF_UNSIGNED_LONG 4",
+    "#undef SIZEOF_VOID_P": "#define SIZEOF_VOID_P 4",
+})
+
 template_rule(
     name = "include_private_hwloc_autogen__config_h",
     src = "include/private/autogen/config.h.in",
     out = "include/private/autogen/config.h",
-    substitutions = if_cuda(
-        _INCLUDE_PRIVATE_HWLOC_AUTOIGEN_CONFIG_H_CUDA_SUBS,
-        if_false = _INCLUDE_PRIVATE_HWLOC_AUTOIGEN_CONFIG_H_LINUX_SUBS,
-    ),
+    substitutions = select({
+        "@//tensorflow_model_server:linux_arm": _INCLUDE_PRIVATE_HWLOC_AUTOIGEN_CONFIG_H_LINUX_ARMV7_SUBS,
+        "//conditions:default": _INCLUDE_PRIVATE_HWLOC_AUTOIGEN_CONFIG_H_LINUX_SUBS,
+    }),
 )

 template_rule(
