--- third_party/hwloc/BUILD.bazel	2019-09-23 13:59:30.284460980 -0700
+++ hwloc.BUILD.bazel	2019-09-23 17:20:04.662594726 -0700
@@ -192,16 +192,24 @@
     "#undef HAVE_CUDA_RUNTIME_API_H": "#undef HAVE_CUDA_RUNTIME_API_H 1",
 }

+_INCLUDE_PRIVATE_HWLOC_AUTOIGEN_CONFIG_H_ARMV7_SUBS = dict()
+_INCLUDE_PRIVATE_HWLOC_AUTOIGEN_CONFIG_H_ARMV7_SUBS.update(_INCLUDE_PRIVATE_HWLOC_AUTOIGEN_CONFIG_H_COMMON_SUBS)
+_INCLUDE_PRIVATE_HWLOC_AUTOIGEN_CONFIG_H_ARMV7_SUBS.update({
+    "#undef HWLOC_SIZEOF_UNSIGNED_LONG": "#define HWLOC_SIZEOF_UNSIGNED_LONG 4",
+    "#undef SIZEOF_UNSIGNED_LONG": "#define SIZEOF_UNSIGNED_LONG 4",
+    "#undef SIZEOF_VOID_P": "#define SIZEOF_VOID_P 4",
+})
+
 _INCLUDE_PRIVATE_HWLOC_AUTOIGEN_CONFIG_H_CUDA_SUBS.update(_INCLUDE_PRIVATE_HWLOC_AUTOIGEN_CONFIG_H_COMMON_SUBS)

 template_rule(
     name = "include_private_hwloc_autogen__config_h",
     src = "include/private/autogen/config.h.in",
     out = "include/private/autogen/config.h",
-    substitutions = if_cuda(
-        _INCLUDE_PRIVATE_HWLOC_AUTOIGEN_CONFIG_H_CUDA_SUBS,
-        if_false = _INCLUDE_PRIVATE_HWLOC_AUTOIGEN_CONFIG_H_COMMON_SUBS,
-    ),
+    substitutions = select({
+        "@//platforms:is_linux_arm": _INCLUDE_PRIVATE_HWLOC_AUTOIGEN_CONFIG_H_ARMV7_SUBS,
+        "//conditions:default": _INCLUDE_PRIVATE_HWLOC_AUTOIGEN_CONFIG_H_COMMON_SUBS,
+    }),
 )

 template_rule(
