--- tensorflow_serving/apis/BUILD	2019-10-17 15:54:03.468824928 -0700
+++ BUILD.apis	2019-10-17 15:56:01.164789619 -0700
@@ -21,7 +21,7 @@
 load("//tensorflow_serving:serving.bzl", "serving_proto_library")
 load("//tensorflow_serving:serving.bzl", "serving_proto_library_py")
 load("//tensorflow_serving:serving.bzl", "serving_go_grpc_library")
-load("@org_tensorflow//tensorflow/core:platform/default/build_config.bzl", "tf_jspb_proto_library", "tf_pyclif_proto_library")
+load("@org_tensorflow//tensorflow/core/platform:default/build_config.bzl", "tf_jspb_proto_library", "tf_pyclif_proto_library")

 serving_proto_library(
     name = "get_model_metadata_proto",
