--- third_party/aws/aws-c-common.bazel	2020-11-23 10:54:13.440058312 -0800
+++ third_party/aws/aws-c-common.bazel.new	2020-12-16 15:34:46.216946839 -0800
@@ -24,7 +24,7 @@
         "@org_tensorflow//tensorflow:linux_ppc64le": glob([
             "source/posix/*.c",
         ]),
-        "@org_tensorflow//tensorflow:raspberry_pi_armeabi": glob([
+        "@org_tensorflow//tensorflow:arm": glob([
             "source/posix/*.c",
         ]),
         "@org_tensorflow//tensorflow:windows": glob([
