--- third_party/aws/BUILD.bazel	2019-07-03 13:06:14.000000000 -0700
+++ aws.BUILD.bazel	2019-07-03 13:09:49.000000000 -0700
@@ -11,24 +11,8 @@

 cc_library(
     name = "aws",
-    srcs = select({
-        "@org_tensorflow//tensorflow:linux_aarch64": glob([
-            "aws-cpp-sdk-core/source/platform/linux-shared/*.cpp",
-        ]),
-        "@org_tensorflow//tensorflow:linux_x86_64": glob([
-            "aws-cpp-sdk-core/source/platform/linux-shared/*.cpp",
-        ]),
-        "@org_tensorflow//tensorflow:macos": glob([
-            "aws-cpp-sdk-core/source/platform/linux-shared/*.cpp",
-        ]),
-        "@org_tensorflow//tensorflow:linux_ppc64le": glob([
-            "aws-cpp-sdk-core/source/platform/linux-shared/*.cpp",
-        ]),
-        "@org_tensorflow//tensorflow:raspberry_pi_armeabi": glob([
-            "aws-cpp-sdk-core/source/platform/linux-shared/*.cpp",
-        ]),
-        "//conditions:default": [],
-    }) + glob([
+    srcs = glob([
+        "aws-cpp-sdk-core/source/platform/linux-shared/*.cpp",
         "aws-cpp-sdk-core/include/**/*.h",
         "aws-cpp-sdk-core/source/*.cpp",
         "aws-cpp-sdk-core/source/auth/**/*.cpp",
